jsonnet/telemeter/rules.libsonnet

```diff
@@ -160,6 +160,56 @@
                 sum(sum_over_time(cluster:capacity_cpu_cores:sum{label_node_role_kubernetes_io = ''}[1h:5m])) by (_id) / scalar(count_over_time(vector(1)[1h:5m]))
               |||,
             },
+            {
+              // OpenShift Cluster effective cores for subscription usage.
+              // This counts both worker nodes and, when the control plane is schedulable, control plane nodes.
+              // Only CoreOS nodes are counted.
+              // 1. x86_64 nodes that show hyperthreading in the telemetry have an accurate cores value in node_role_os_version_machine:cpu_capacity_cores:sum.
+              // 2. x86_64 nodes that are do not show hyperthreading and are bare metal have an accurate cores value in node_role_os_version_machine:cpu_capacity_cores:sum.
+              // 3. x86_64 nodes that are do not show hyperthreading and are not bare metal need the cores value adjusted to account for 2 threads per core (* 0.5).
+              // 4. Other CPU architectures are assumed to have accurate values in node_role_os_version_machine:cpu_capacity_cores:sum.
+              record: 'cluster:usage:workload:capacity_effective_cpu_cores',
+              expr: |||
+                # worker ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="true"}) or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
+
+                +
+
+                # worker virt non-ht amd64
+                ((sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="false"}) * on(_id) group(cluster_infrastructure_provider{type!="BareMetal"}) by (_id)) / 2.0 or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
+
+                +
+
+                # worker phys non-ht amd64
+                (sum by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum{label_node_openshift_io_os_id="rhcos",label_node_role_kubernetes_io_infra!="true",label_node_role_kubernetes_io_master!="true",label_kubernetes_io_arch="amd64",label_node_hyperthread_enabled="false"})* on(_id) cluster_infrastructure_provider{type="BareMetal"} or (0 * max by (_id) (node_role_os_version_machine:cpu_capacity_cores:sum)))
```

Comment:
I can see clusters where cluster_infrastructure_provider has `type="BareMetal"` but all nodes are reported as vmware in `cluster:virt_platform_nodes:sum`. IIUC the query will not divide node_role_os_version_machine:cpu_capacity_cores:sum by 2 in this case.

Reply:
@simonpasquier oof, that's a gap for sure. I'm going to discuss with stakeholders, in order to address properly we might actually need an in-cluster monitoring change.

Reply:
@simonpasquier we're currently thinking we'll simplify a bit and handle those use cases by not attempting to distinguish physical and virtual nodes. If hyperthreading isn't detected, then we'll divide by 2. This seems reasonable, as OCP subs use as a unit of measure "Cores or vCPUs" with a ratio of 2 vCPUs per core. In practice, a 4 CPU physical node, backed by 4 physical cores, would be counted as "2 cores". (Note if this same node was deployed on the same hardware, but on a VM, it'd also be counted as "2 cores"). If hyperthreading were enabled for that same hardware, it'd be reported as 4 cores, and a VM would have access to 8 vCPUs. So it's at least consistent.